components/input-group/src/InputGroup.js

```diff
@@ -0,0 +1,85 @@
+/** @jsx jsx */
+
+import React, { Children, cloneElement } from 'react';
+import PropTypes from 'prop-types';
+import { jsx } from '@westpac/core';
+
+// ==============================
+// Component
+// ==============================
+
+/**
+ * Input Group: Styled input fields with attached addons. Addons can be text ($, %, .00 etc) or form controls (buttons or select inputs).
+ */
+export const InputGroup = ({ size, children, ...props }) => {
+	const childrenWithProps = Children.map(children, child => cloneElement(child, { size }));
+
+	const style = {
+		display: 'flex',
+
+		select: {
+			width: 'auto',
+		},
+
+		'select:not(:first-child)': {
+			marginLeft: '-1px',
+		},
+
+		'select:not(:last-child)': {
+			marginRight: '-1px',
+		},
+
+		'input:not(:first-child), span:not(:first-child), button:not(:first-child), select:not(:first-child)': {
```

Comment:
should change the `:first-child` selectors to `:first-of-type` selectors as per react server side rendering recommendation

Reply:
`:first-of-type` does not work in that case. I need to find another way (not css selector) to target these elements...
We do not use SSR so should we ignore? https://github.com/emotion-js/emotion/issues/1105